File: Calendar.java
import java.util.*;
public class Calendar {
    int start(int y){
        int mm=0,l=0;
        for(int h=1900;h<y;h++){
            if((h%4==0&&h%100!=0)||(h%400==0))
            mm+=366;
            else
            mm+=365;
        }
        l=(mm+1);
        return l;
    }
    int leap(int y,int i){
        int b;
        if((i<=7&&i%2==1)||(i>=8&&i%2==0))
        b=31;
        else if(((y%4==0&&y%100!=0)||(y%400==0))&&i==2)
        b=29;
        else if(i==2)
        b=28;
        else
        b=30;
        return b;
    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        Calendar ob=new Calendar();
        System.out.println("Enter year");
        int y=sc.nextInt(),i,j,k,l,b=0;
        l=ob.start(y);
        System.out.println("Enter the month number");
        int month=sc.nextInt();
        String m[]={"JANUARY","FEBRUARY","MARCH","APRIL","MAY","JUNE","JULY","AUGUST","SEPTEMBER","OCTOBER","NOVEMBER","DECEMBER"};
        System.out.println(m[month-1]);
        System.out.println("S\tM\tT\tW\tT\tF\tS");
        for(i=1;i<month;i++){
            b=ob.leap(y,i);
            l+=b;
        }
        b=ob.leap(y, i);
        l=l%7+1;
        for(j=1;j<l;j++)
        System.out.print(" \t");
        for(k=1;k<=b;k++,j++){
            System.out.print(k+"\t");
            l++;
            if(l==8||j%7==0){
                System.out.println();
                l=1;
            }
        }
    }
}

File: Calendar.py
def start(y):
    mm=0
    for i in range (1900,y,1) :
        if ((i%4==0 and i%100!=0) or i%400==0):
            mm+=366
        else:
            mm+=365
    p=(mm+1)%7
    return p


y=int(input("Enter year\n"))
b=0
l=start(y)+1
m=("JANUARY","FEBRUARY","MARCH","APRIL","MAY","JUNE","JULY","AUGUST","SEPTEMBER","OCTOBER","NOVEMBER","DECEMBER")
for i in range(1,13,1):
    if((i<=7 and i%2==1)or(i>=8 and i%2==0)):
        b=31
    elif(((y%4==0 and y%100!=0)or(y%400==0))and i==2):
        b=29
    elif i==2:
        b=28
    else:
        b=30
    print(m[i-1],": ")
    print("S\tM\tT\tW\tT\tF\tS")
    k=0
    for k in range(1,l,1):
        print("\t",end="")
    k+=1
    for j in range(1,b+1,1):
        print(j ,"\t", end="")
        l+=1
        if l==8 or k%7==0:
            print()
            l=1
        k+=1
    print()